typing/printtyp.ml

```diff
@@ -1124,14 +1162,16 @@ let rec tree_of_typexp mode ty =
           (* Make the names delayed, so that the real type is
              printed once when used as proxy *)
           List.iter add_delayed tyl;
-          let tl = List.map (Names.name_of_type Names.new_name) tyl in
+          let tl =
+            List.map (Names.name_of_type ~non_gen:false Names.new_name) tyl
+          in
           let tr = Otyp_poly (tl, tree_of_typexp mode ty) in
           (* Forget names when we leave scope *)
           Names.remove_names tyl;
           delayed := old_delayed; tr
         end
     | Tunivar _ ->
-        Otyp_var (false, Names.name_of_type Names.new_name tty)
+        Otyp_var (false, Names.name_of_type ~non_gen:false Names.new_name tty)
```

Comment:
I am not sure if the character shifting heuristics work well in this case. If the user code contains
`<m: 'a. 'a -> 'a >` and the `'a` name is already taken, I have the impression that we may want to keep the alternative name as close as possible as `'a` (aka `'a0`  in this case).